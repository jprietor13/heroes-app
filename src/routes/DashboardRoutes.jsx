import React from "react";
import { Route, Routes } from "react-router-dom";
import { Navbar } from "../components/ui/Navbar";
import { MarvelScreen } from "../components/marvel/MarvelScreen";
import { DcScreen } from "../components/dc/DcScreen";
import { SearchScreen } from "../components/search/SearchScreen";
import { HeroScreen } from "../components/hero/HeroScreen";

export const DashboardRoutes = () => {
  return (
    <>
      <Navbar />
      <div className="container mt-5">
      <Routes>
        <Route exact path="/" element={<MarvelScreen />} />
        <Route exact path="/marvel" element={<MarvelScreen />} />
        <Route exact path="/dc" element={<DcScreen />} />
        <Route exact path="/search" element={<SearchScreen />} />
        <Route exact path="/hero/:idHero" element={ <HeroScreen /> }/>
      </Routes>
      </div>
    </>
  );
};
