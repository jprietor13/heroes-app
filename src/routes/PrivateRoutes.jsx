import React, { useContext } from 'react'
import { AuthContext } from '../auth/authContext'
import { Navigate, useLocation } from 'react-router-dom'

export const PrivateRoutes = ({ children }) => {

  const { user } = useContext(AuthContext)

  const { search, pathname } = useLocation();

  localStorage.setItem('lastPath', pathname + search)

  return (
    user.logged ? children : <Navigate to="/login" /> 
  )
}
