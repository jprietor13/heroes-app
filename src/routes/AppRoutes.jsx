import React from 'react'
import { BrowserRouter, Routes, Route } from 'react-router-dom'
import { LoginScreen } from '../components/login/LoginScreen'
import { DashboardRoutes } from './DashboardRoutes'


export const AppRoutes = () => {
  return (
    <BrowserRouter>
      <Routes>
        <Route exact path="/login" element={ <LoginScreen /> }/>
        <Route exact path="*" element={ <DashboardRoutes /> } />
      </Routes>
    </BrowserRouter>
  )
}
