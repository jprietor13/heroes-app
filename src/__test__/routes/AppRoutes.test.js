import { AppRoutes } from "../../routes/AppRoutes";
import { mount } from 'enzyme'
import { AuthContext } from "../../auth/authContext";

describe('Pruebas en <AppRoutes />', () => {

 

  it("Debe de mostrar el login si el usuario no esta autenticado", () => {

    const contextValue = {
      user: {
        logged: false
      }
    }

    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <AppRoutes />
      </AuthContext.Provider>
    )
   
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('h1').text().trim()).toBe('Login')

  })

  it("Debe de mostrar el componente de marvel cuando el usuario esta autenticado", () => {

    const contextValue = {
      user: {
        logged: true,
        name: 'Juan Prieto'
      }
    }

    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <AppRoutes />
      </AuthContext.Provider>
    )
   
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('.navbar').exists()).toBe(true)

  })

})