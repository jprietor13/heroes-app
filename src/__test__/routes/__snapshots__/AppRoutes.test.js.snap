// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRoutes /> Debe de mostrar el login si el usuario no esta autenticado 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login
  </h1>
  <hr />
  <input
    autoComplete="off"
    className="form-control"
    name="userName"
    onChange={[Function]}
    placeholder="Nombre Usuario"
    type="text"
    value=""
  />
  <button
    className="btn btn-primary mt-2"
    onClick={[Function]}
    type="button"
  >
    Login
  </button>
</div>
`;
